src/TypeSpec.Extension/Emitter.Csharp/src/emitter.ts

```diff
@@ -28,107 +28,120 @@ export async function $onEmit(context: EmitContext<AzureNetEmitterOptions>) {
     /* set the loglevel. */
     Logger.initialize(program, options.logLevel ?? LoggerLevel.INFO);
 
-    context.options.skipSDKGeneration = true;
-    if (
-        context.emitterOutputDir &&
-        path.basename(context.emitterOutputDir) === "src"
-    ) {
-        context.emitterOutputDir = path.dirname(context.emitterOutputDir);
-    }
-    Logger.getInstance().info("Starting Microsoft Generator Csharp emitter.");
-    setSDKContextOptions(azureSDKContextOptions);
-    await $OnMGCEmit(context);
-    const outputFolder = resolveOutputFolder(context);
-    const isSrcFolder = path.basename(outputFolder) === "src";
-    const generatedFolder = isSrcFolder
-        ? resolvePath(outputFolder, "Generated")
-        : resolvePath(outputFolder, "src", "Generated");
-    if (options.skipSDKGeneration !== true) {
-        const configurationFilePath = resolvePath(
-            outputFolder,
-            configurationFileName
-        );
-        const configurations = JSON.parse(
-            fs.readFileSync(configurationFilePath, "utf-8")
-        );
-        //resolve shared folders based on generator path override
-        const resolvedSharedFolders: string[] = [];
-        const sharedFolders = [
-            resolvePath(options.csharpGeneratorPath, "..", "Generator.Shared"),
-            resolvePath(options.csharpGeneratorPath, "..", "Azure.Core.Shared")
-        ];
-        for (const sharedFolder of sharedFolders) {
-            resolvedSharedFolders.push(
-                path
-                    .relative(generatedFolder, sharedFolder)
-                    .replaceAll("\\", "/")
-            );
+    if (!program.compilerOptions.noEmit && !program.hasError()) {
```

Comment:
can we revert the if? It would be more clear and reduce the amount of changes here.